pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -74,30 +73,39 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 		return nil
 	}
 
-	// we can always allow the upgrade if there isn't a version already installed
-	if len(cv.Status.History) == 0 {
-		klog.V(2).Infof("Precondition %s passed: no release history.", pf.Name())
-		return nil
+	currentVersion, err := semver.Parse(cv.Status.Desired.Version)
+	if err != nil {
+		return &precondition.Error{
+			Nested:             err,
+			Reason:             "InvalidCurrentVersion",
+			Message:            err.Error(),
+			Name:               pf.Name(),
+			NonBlockingWarning: true, // do not block on issues that require an update to fix
+		}
 	}
 
-	currentVersion := GetCurrentVersion(cv.Status.History)
-	currentMinor := GetEffectiveMinor(currentVersion)
-	desiredMinor := GetEffectiveMinor(releaseContext.DesiredVersion)
-	klog.V(2).Infof("currentMinor %s releaseContext.DesiredVersion %s desiredMinor %s", currentMinor, releaseContext.DesiredVersion, desiredMinor)
+	targetVersion, err := semver.Parse(releaseContext.DesiredVersion)
+	if err != nil {
+		return &precondition.Error{
+			Nested:  err,
+			Reason:  "InvalidDesiredVersion",
+			Message: err.Error(),
+			Name:    pf.Name(),
+		}
+	}
 
-	// if there is no difference in the minor version (4.y.z where 4.y is the same for current and desired), then we can still upgrade
-	// if no cluster overrides have been set
-	if !minorVersionUpgrade(currentMinor, desiredMinor) {
-		klog.V(2).Infof("Precondition %q passed: minor from the target %s is not a minor version update from the current %s.%s.", pf.Name(), releaseContext.DesiredVersion, currentVersion, currentMinor)
+	klog.V(2).Infof("The current version is %s parsed from %s and the target version is %s parsed from %s", currentVersion.String(), cv.Status.Desired.Version, targetVersion.String(), releaseContext.DesiredVersion)
+	if targetVersion.Major == currentVersion.Major && targetVersion.Minor == currentVersion.Minor {
+		// When Upgradeable==False, a patch level update with the same minor level is allowed unless overrides are set
 		if condition := ClusterVersionOverridesCondition(cv); condition != nil {
-			klog.V(2).Infof("Update from %s to %s blocked by %s: %s", currentVersion, releaseContext.DesiredVersion, condition.Reason, condition.Message)
-
+			klog.V(2).Infof("Update from to %s is blocked by %s: %s", targetVersion.String(), condition.Reason, condition.Message)
```

Comment:
`Update from to %s` should either drop `from` or (my weak preference) go back to the previous `... from %s to %s`.  If you don't like the `Update` as being confusing between "where the cluster had been trying to go" and "where the cluster may or may not have actually been", perhaps `Retarget ...`?